lib/matplotlib/backends/qt_compat.py

```diff
@@ -51,20 +60,34 @@
         QT_API = None
 # A non-Qt backend was selected but we still got there (possible, e.g., when
 # fully manually embedding Matplotlib in a Qt app without using pyplot).
+elif QT_API_ENV is None:
+    QT_API = None
 else:
     try:
         QT_API = _ETS[QT_API_ENV]
     except KeyError as err:
         raise RuntimeError(
             "The environment variable QT_API has the unrecognized value {!r};"
-            "valid values are 'pyqt5', and 'pyside2'") from err
+            "valid values are {}".format(
+                QT_API, ", ".join(map(repr, _ETS)))) from None
```

Comment:
this should be QT_API_ENV (QT_API is undefined at this point)

Reply:
Now looks like:

```
In [1]: import matplotlib.backends.qt_compat
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-1-9f683fba02b3> in <module>
----> 1 import matplotlib.backends.qt_compat

~/source/p/matplotlib/matplotlib/lib/matplotlib/backends/qt_compat.py in <module>
     67         QT_API = _ETS[QT_API_ENV]
     68     except KeyError as err:
---> 69         raise RuntimeError(
     70             "The environment variable QT_API has the unrecognized value "
     71             f"{QT_API_ENV!r}; "

RuntimeError: The environment variable QT_API has the unrecognized value 'boo'; valid values are {'pyside6', 'pyqt5', 'pyqt6', 'pyside2'}

In [2]: 
```